country-a-service/epps-service/src/main/java/dk/sundhedsdatastyrelsen/ncpeh/service/mapping/EPrescriptionMapper.java

```diff
@@ -58,14 +42,60 @@ public static DocumentAssociationForEPrescriptionDocumentMetadataDto mapMeta(Str
         }
     }
 
-    private static EPrescriptionDocumentMetadataDto generateMeta(String patientId, EPrescriptionL3 model, String documentId) {
+    private static EPrescriptionDocumentMetadataDto generateMeta(String patientId, EPrescriptionL3 model, DocumentLevel documentLevel) {
+        if (!documentLevel.equals(DocumentLevel.LEVEL1) && !documentLevel.equals(DocumentLevel.LEVEL3)) {
+            throw new IllegalArgumentException("Does not support documentLevel: " + documentLevel.toString());
+        }
+
+        String documentId = switch (documentLevel) {
+            case DocumentLevel.LEVEL1 ->
+                EPrescriptionDocumentIdMapper.level1DocumentId(model.getPrescriptionId().getExtension());
+            case DocumentLevel.LEVEL3 ->
+                EPrescriptionDocumentIdMapper.level3DocumentId(model.getPrescriptionId().getExtension());
+        };
+
+        DocumentFormatDto documentFormat = switch (documentLevel) {
+            case DocumentLevel.LEVEL1 -> DocumentFormatDto.PDF;
+            case DocumentLevel.LEVEL3 -> DocumentFormatDto.XML;
+        };
+
         var meta = new EPrescriptionDocumentMetadataDto(documentId);
         meta.setPatientId(patientId);
         meta.setEffectiveTime(model.getEffectiveTimeOffsetDateTime());
         meta.setRepositoryId(Oid.DK_EPRESCRIPTION_REPOSITORY_ID.value);
         meta.setAuthor(model.getAuthor().getName().getFullName());
         meta.setTitle(model.getTitle());
         meta.setDescription(model.getIndicationText());
+        meta.setProductCode(model.getProduct().getPackageCode().getCode());
+        meta.setAtcCode(model.getProduct().getAtcCode().getCode());
+        meta.setAtcName(model.getProduct().getAtcCode().getDisplayName());
+        meta.setClassCode(ClassCodeDto._57833_6); // Prescription for medication
+        meta.setDispensable(true); //This should always be true, we don't return non-dispensable prescriptions
+        meta.setFormat(documentFormat);
+        meta.setLanguage("da-DK"); //We always include danish text in free-text, so
+        meta.setProductCode(model.getProduct()
+            .getPackageCode()
+            .getCode()); //Varenummer
+        meta.setProductName(model.getProduct().getName());
+        meta.setDoseFormCode(model.getProduct()
+            .getFormCode()
+            .getCode());
+        meta.setDoseFormName(model.getProduct().getFormCode().getDisplayName());
+        meta.setConfidentiality(new ConfidentialityMetadataDto().confidentialityCode("N")
+            .confidentialityDisplay("Normal"));
+        meta.setStrength(model.getProduct().getStrength());
+
+        //The following data is set to this by convention to indicate a document generated on-demand
+        // https://profiles.ihe.net/ITI/TF/Volume2/ITI-38.html
+        meta.setHash("da39a3ee5e6b4b0d3255bfef95601890afd80709"); //SHA1 hash of a zero length file
+        meta.setSize(0L);
+
+        // TODO Missing metadata fields as of now
```

Comment:
it can be hard to know when "now" is
```suggestion
        // TODO Missing metadata fields as of 2025-04-01
```